quantum/process_keycode/process_auto_shift.c

```diff
@@ -75,20 +105,28 @@ uint16_t get_autoshift_timeout(void) { return autoshift_timeout; }
 void set_autoshift_timeout(uint16_t timeout) { autoshift_timeout = timeout; }
 
 bool process_auto_shift(uint16_t keycode, keyrecord_t *record) {
+    autoshift_flush();
     if (record->event.pressed) {
+#    ifdef PERMISSIVE_HOLD
```

Comment:
```suggestion
#    if defined(PERMISSIVE_HOLD) || defined(PERMISSIVE_HOLD_PER_KEY)
```

To account for pending [PR #7994](https://github.com/qmk/qmk_firmware/pull/7994)